src/model/Weather.js

```diff
@@ -165,9 +166,11 @@ export default class Weather{
     set specialWeather(specialWeather){
         if(Array.isArray(specialWeather)){
             let keys = Object.keys(Weather.SPECIAL_WEATHER);
+            // CODE_REVIEW tutaj zdecydowanie for...of by się nadało. Tak samo .forEach().
             for(let i=0;i<specialWeather.length;i++){
                 let j = 0;
                 let legalValue = false;
+                // CODE_REVIEW Dlaczego nie pętla for albo for...of?
```

Comment:
To trochę kwestia osobistych preferencji jeśli o konwencję chodzi - sprowadza się tu do koncepcji, żeby pętle typu "for" stosować tam, gdzie przewidywanym warunkiem przerwania pętli jest przeiterowanie wszystkiego, co było zaplanowane (kolekcja się skończyła, indeks urósł do zadanej wartości etc.). Jeśli podstawowym warunkiem przerwania pętli ma jednak być spełnienie jakiegoś innego warunku (znalezienie szukanej wartości, na przykład) to lepszym wyborem jest "while" (bo z założenia iterujemy, aż spełnimy warunek - dotarcie "do końca" jest sytuacją szczególną jeśli nie wręcz błędną).
Technicznych powodów po temu nie ma, bo for może mieć bardziej skomplikowany warunek wyjścia, nie mówiąc o tym, że istnieje break; na tym świecie, jak ktoś bardzo potrzebuje ;)